AK/QuickSort.h

```diff
@@ -14,10 +15,21 @@ namespace AK {
  * pivot quick_sort below. The other quick_sort below should only be used when
  * you are stuck with simple iterators to a container and you don't have access
  * to the container itself.
+ *
+ * We use a cutoff to insertion sort for partitions of size 7 or smaller.
+ * The idea is to avoid recursion for small partitions.
+ * The value 7 here is a magic number. According to princeton's CS algorithm class
+ * a value between 5 and 15 should work well in most situations:
+ * https://algs4.cs.princeton.edu/23quicksort/
  */
 template<typename Collection, typename LessThan>
 void dual_pivot_quick_sort(Collection& col, int start, int end, LessThan less_than)
 {
+    if ((end + 1) - start <= 7) {
+        AK::insertion_sort(col, start, end, less_than);
```

Comment:
No `AK` prefix anywhere*, especially not in AK itself

\*name conflicts excluded